torch/_dynamo/variables/functions.py

```diff
@@ -601,3 +601,30 @@ def call_function(
                 f"CollectiveFunctionRewriteVariable can't support async_op=True for {self.orig_fn}"
             )
         return super().call_function(tx, args, kwargs)
+
+
+class FunctoolsPartialVariable(VariableTracker):
+    def __init__(self, func, args, keywords, **kwargs):
+        super().__init__(**kwargs)
+        self.func = func
+        assert isinstance(args, list)
+        self.args = args
+        assert isinstance(keywords, dict)
+        self.keywords = keywords
+
+        self.guards.update(VariableTracker.propagate(func)["guards"])
+        for arg in args:
+            self.guards.update(VariableTracker.propagate(arg)["guards"])
+        for val in keywords.values():
+            self.guards.update(VariableTracker.propagate(val)["guards"])
```

Comment:
Why not just:
```suggestion
        self.guards.update(VariableTracker.propagate(func, args, keywords)["guards"])
```

Reply:
You cant propagate lists and dicts, and I didn't see a reason to go through making a ListVariable/ConstDictVariable just for these.